### PR TITLE
fix: update 0006 async migration where clause

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

The backfill from the 0006 async migration was inconsistent to varying degrees on Cloud and for self-hosted users.

This is most likely caused by the fact that columns are updated individually + we do complex dictionary fetches leading to varying times to update a column.

The main suspicion is that by having a where clause checking for empty person IDs we failed to update the other columns when the person_id was updated first.

## Changes

- Updates the WHERE clause to be timestamp-based (for now). This is fine for us but won't be great for self-hosted users. I'm exploring ways to make it work for them
- Adds a flag `ALLOW_EXPERIMENTAL_ASYNC_MIGRATIONS`. I want to test this on the playground and then on Cloud and this is a way to merge this code while safeguarding against people who pin latest
- Adds new parameters to the migration - timestamp upper and lower bound + team ID

## How did you test this code?

Current tests all pass. I'll write some more tests for the new functionality but submitting this already to Team West has enough time to look over it

### CODE REVIEW

File: posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py [hunk 1]
```diff
@@ -75,9 +76,19 @@ class Migration(AsyncMigrationDefinition):
...
+            None,
+            "The team_id of team to run backfill for. If unset the backfill will run for all teams.",
+            str,
+        ),
     }
 
     def precheck(self):
+        # Used to guard against self-hosted users running on `latest` while we make tweaks to the migration
+        if not settings.TEST and get_instance_setting("ALLOW_EXPERIMENTAL_ASYNC_MIGRATIONS"):
+            return (False, "ALLOW_EXPERIMENTAL_ASYNC_MIGRATIONS is set to False")
```
Comment (EDsCODE) [inline posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py:91]: Is this intended? `ALLOW_EXPERIMENTAL_ASYNC_MIGRATIONS` would be true if this is reached

**Review by EDsCODE — COMMENTED** [review]
Looks good for a test

File: posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py (hunk 1 above)
Comment (yakkomajuri) [reply posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py:91]: ah oops I reused the `not` when I popped in `settings.TEST` - will update

good catch!

**Comment by yakkomajuri** [comment]
Tests pass in isolation but not when the full suite runs - debugging interactions

**Pull request merged**